sessions/return-oriented-programming/README.md

```diff
@@ -0,0 +1,810 @@
+# Return Oriented Programming
+
+## Recap - ASLR
+ASLR is not the only feature that prevents the compiler and the linker from
+solving some relocations before the binary is actually running. Shared libraries
+can also be combined in different ways, so the first time you actually know the
+address of a shared library is while the loader is running. The ASLR feature is
+orthogonal to this - the loader could choose to assign the addresses to
+libraries in a round-robin fashion, or could use ASLR to assign them randomly.
+
+Of course, we might be inclined to have the loader simply fix all relocations in
+the code section after it loaded the libraries, but this breaks the memory
+access protection of the `.text` section, which should only be **readable** and
+**executable**.
+
+## Solution - GOT and PLT
+In order to solve this issue, we need another level of indirection by which all
+accesses to symbols located in shared libraries will read the actual address
+from a table, called the **Global Offset Table (`.got`)**, at runtime. The one

```

Comment:
Explain a bit more clearly? the `by which` part extends the sentence way too much and becomes a bit hard to follow. Maybe split the sentence into two?